smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/WaiterGenerator.java

```diff
@@ -81,16 +83,34 @@ private void generateWaiter() {
                             waiter.getMaxDelay());
             writer.write("return createWaiter({...serviceDefaults, ...params}, input, checkState);");
         });
+
+        // generates WaitUtil....
+        writer.writeDocs(waiter.getDocumentation().orElse("") + " \n"
+                + " @param params : Waiter configuration options.\n"
+                + " @param input : the input to " + operationSymbol.getName() + " for polling.");
+        writer.openBlock("export const waitUntil$L = async (params: WaiterConfiguration<$T>, input: $T): "
+                + "Promise<WaiterResult> => {", "}", waiterName, serviceSymbol, inputSymbol, () -> {
+            writer.write("const serviceDefaults = { minDelay: $L, maxDelay: $L };", waiter.getMinDelay(),
+                    waiter.getMaxDelay());
+            writer.write("const result = await createWaiter({...serviceDefaults, ...params}, input, checkState);");
+            writer.openBlock("if (result.state !== WaiterState.SUCCESS) {", "}", () -> {
+                writer.write("throw result;");
+            });
```

Comment:
AbortError needs to be thrown in `ABORTED` state as per https://github.com/aws/aws-sdk-js-v3/issues/2316
```suggestion
            writer.openBlock("if (result.state === WaiterState.ABORTED) {", "}", () -> {
                writer.write("throw Object.assign(new Error(\"Request aborted\"),"
                  + " {name: \"AbortError\", reason: result.reason});");
            });
            writer.openBlock("else if (result.state !== WaiterState.SUCCESS) {", "}", () -> {
                writer.write("throw Object.assign(new Error(result.state), { reason: result.reason });");
            });
```